userns.go

```diff
@@ -1,18 +1,21 @@
+//go:build linux
+
```

Comment:
Just to make sure does this still build the freebsd build in podman?

Another thing if we now that this will only ever be used on linux we can rename userns.go to userns_linux.go which I think is a bit better, also for the test userns_linux_test.go

Reply:
Should be fine, FreeBSD doesn't support user namespaces last I checked.